test/views/ViewArrivalsSpec.scala

```diff
@@ -75,9 +77,48 @@ class ViewArrivalsSpec extends SingleViewSpec("viewArrivals.njk") with Generator
     ls.eq(5).text() mustBe "11 August 2020"
   }
 
+  val rows: Elements = doc.getElementsByAttributeValue("role", "row")
+
   "generate a row for each arrival" in {
-    val ls: Elements = doc.getElementsByAttributeValue("role", "row")
+    rows.size() mustEqual 7
+  }
 
-    ls.size() mustEqual 7
+  "display rows in correct (sorted) order" in {
+    rows.toList.zipWithIndex.forEach {
+      x =>
+        val cells: Elements = x._1.getElementsByAttributeValue("role", "cell")
+
+        cells.get(0).text() mustBe "viewArrivalNotifications.table.updated " + (x._2 match {
+          case 0 => "6:06am"
+          case 1 => "5:05am"
+          case 2 => "4:04am"
+          case 3 => "3:03am"
+          case 4 => "2:02am"
+          case 5 => "1:01am"
+          case 6 => "1:00am"
+        })
+    }
   }
+
+  "display correct data in each row" in {
+    rows.toList.zipWithIndex.forEach {
+      x =>
+        val cells: Elements = x._1.getElementsByAttributeValue("role", "cell")
+
+        cells.get(1).ownText() mustBe viewMovements(x._2).movementReferenceNumber
+        cells.get(1).text() mustBe s"viewArrivalNotifications.table.mrn ${viewMovements(x._2).movementReferenceNumber}"
+
+        cells.get(2).ownText() mustBe viewMovements(x._2).status
+        cells.get(2).text() mustBe s"viewArrivalNotifications.table.status ${viewMovements(x._2).status}"
+
+        val actions = cells.get(3).getElementsByTag("a")
+        actions.toList.zipWithIndex.forEach {
+          y =>
+            y._1.text() mustBe s"${viewMovements(x._2).actions(y._2).key} ${s"${viewMovements(x._2).actions(y._2).key}.hidden"}"
+            y._1.attr("id") mustBe s"${viewMovements(x._2).actions(y._2).key}-${viewMovements(x._2).movementReferenceNumber}"
+            y._1.attr("href") mustBe viewMovements(x._2).actions(y._2).href
+        }
+    }
+  }
```

Comment:
I think this test might need to be refactored, it's adding a lot of good coverage but isn't clear on how it's doing it